### PR TITLE
Invalid QR dialog wrong button functions (EXPOSUREAPP-7341)

### PR DESCRIPTION
**Pull request by SamuraiKek**
Fixes an issue where the back button dismissed the dialog but the cancel button on the dialog would navigate you back to the previous screen. 
Now pressing the back button or the cancel button on the dialog will have the same effect.

Test by scanning an invalid corona test QR code.

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/DialogHelper.kt [hunk 1]
```diff
@@ -92,6 +92,7 @@ object DialogHelper {
                     ) { _, _ ->
                         dialogInstance.negativeButtonFunction()
                     }
+                    setOnCancelListener { dialogInstance.negativeButtonFunction() }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/DialogHelper.kt:95]: This affecting every dialog that uses `DialogHelper ` , right?
Can you make that change exclusively for the dialog in `SubmissionQRCodeScanFragment `?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/DialogHelper.kt [hunk 2]
```diff
@@ -93,6 +96,9 @@ object DialogHelper {
                         dialogInstance.negativeButtonFunction()
                     }
                 }
+                if (dialogInstance.cancelFunction != null) {
+                    setOnCancelListener { dialogInstance.cancelFunction() }
+                }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/DialogHelper.kt:101]: IDE is suggesting to simplify that to 
```kotlin
   setOnCancelListener { dialogInstance.cancelFunction() }
```

**Comment by mtwalli** [comment]
Fix is not working as expected, please double check?

**Comment by SamuraiKek** [comment]
> Fix is not working as expected, please double check?

Latest changes should address this.

**Review by AlexanderAlferov — APPROVED** [review]
LGTM 🚀

**Review by jurajkusnier — APPROVED** [review]

**Pull request merged**